### PR TITLE
feat(grafana json): update grafana json for nginx shared dict in prometheus

### PR DESCRIPTION
**Pull request by ccxhwmy**
update grafana json for nginx shared dict in prometheus

### Description
update grafana json to add nginx shared dict, which default config is `internal-status|upstream-healthcheck|worker-events|lrucache-lock|balancer-ewma|balancer-ewma-locks|balancer-ewma-last-touched-at|etcd-cluster-health-check`.

Fixes # (issue)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by soulbird** [comment]
image
We can't see the change of `apisix_shared_dict_free_space_bytes` through the ordinate, can you adjust the unit? That would be more intuitive. Please note that we cannot determine the size of the shared memory in the actual environment, it may be 1MB or 100MB, so pay attention to the generality of the unit.

File: docs/assets/other/json/apisix-grafana-dashboard.json [hunk 1]
```diff
@@ -1622,6 +1622,216 @@
...
+      "pointradius": 2,
+      "points": false,
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "expr": "sum(apisix_shared_dict_free_space_bytes{name=~\"internal-status|upstream-healthcheck|worker-events|lrucache-lock|balancer-ewma|balancer-ewma-locks|balancer-ewma-last-touched-at|etcd-cluster-health-check\", instance=~\"$instance\"}) by (name)",
```
Comment (spacewander) [inline docs/assets/other/json/apisix-grafana-dashboard.json:1783]: We don't need to hardcode the name of each shdict, right?

BTW, we should provide an available percent (free / capacity) for each shdict than the sum. The sum is not helpful.

Comment (ccxhwmy) [reply docs/assets/other/json/apisix-grafana-dashboard.json:1783]: > We don't need to hardcode the name of each shdict, right?
> 
> BTW, we should provide an available percent (free / capacity) for each shdict than the sum. The sum is not helpful.

I tryed to do it but failed, would you give me some suggestion(like use which function in grafana) in caculate the percent (free/capacity) for each shdict, thank you very much ! : )

Comment (spacewander) [reply docs/assets/other/json/apisix-grafana-dashboard.json:1783]: You can learn about the syntax of PromQL and try for it. AFAIK, a division between free and capacity is enough.

File: docs/assets/other/json/apisix-grafana-dashboard.json [hunk 2]
```diff
@@ -1622,6 +1622,111 @@
...
+      "pointradius": 2,
+      "points": false,
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "expr": "(apisix_shared_dict_free_space_bytes * 100) / on (name) apisix_shared_dict_capacity_bytes",
```
Comment (spacewander) [inline docs/assets/other/json/apisix-grafana-dashboard.json:1678]: If you are worried about showing instances will make the graph too complex, maybe we can add a comment that people can modify this query to show per-instance graph?

**Review by spacewander — APPROVED** [review]
Anyway, LGTM

**Review by soulbird — APPROVED** [review]

File: docs/assets/other/json/apisix-grafana-dashboard.json (hunk 2 above)
Comment (ccxhwmy) [reply docs/assets/other/json/apisix-grafana-dashboard.json:1678]: > If you are worried about showing instances will make the graph too complex, maybe we can add a comment that people can modify this query to show per-instance graph?

Does you mean add a comment like `# we can modify the expr to show per-instance graph` in json file?

**Pull request merged**